tests/storage/reservation.go

```diff
@@ -0,0 +1,295 @@
+			"backstores/fileio",
+			"create", backendDisk, "/disks/disk.img", "1G"})
+		executeTargetCli(podName, []string{
+			"loopback/", "create", naa})
+		// Create LUN
+		executeTargetCli(podName, []string{
+			fmt.Sprintf("loopback/naa.%s/luns", naa),
+			"create",
+			fmt.Sprintf("/backstores/fileio/%s", backendDisk)})
+	}
+
+	// findSCSIdisk returns the first scsi disk that correspond to the model. With targetcli the model name correspond to the name of the storage backend.
+	// Example:
+	// $ lsblk --scsi -o NAME,MODEL -p -n
+	// /dev/sda disk1
+	findSCSIdisk := func(podName string, model string) string {
+		var device string
+		pod, err := virtClient.CoreV1().Pods(testsuite.NamespacePrivileged).Get(context.Background(), podName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		stdout, stderr, err := exec.ExecuteCommandOnPodWithResults(virtClient, pod, "targetcli",
+			[]string{"/bin/lsblk", "--scsi", "-o", "NAME,MODEL", "-p", "-n"})
+		By(fmt.Sprintf("targetcli: stdout: %v stderr: %v", stdout, stderr))
+		Expect(err).ToNot(HaveOccurred())
+		lines := strings.Split(stdout, "\n")
+		for _, line := range lines {
+			if strings.Contains(line, model) {
+				line = strings.TrimSpace(line)
+				disk := strings.Split(line, " ")
+				if len(disk) < 1 {
+					continue
+				}
+				device = disk[0]
+				break
+			}
+		}
+		return device
+
+	}
+
+	checkResultCommand := func(vmi *v1.VirtualMachineInstance, cmd, output string) bool {
+		res, err := console.SafeExpectBatchWithResponse(vmi, []expect.Batcher{
+			&expect.BSnd{S: fmt.Sprintf("%s\n", cmd)},
+			&expect.BExp{R: console.PromptExpression},
+		}, 20)
+		Expect(err).ToNot(HaveOccurred())
+		return strings.Contains(res[0].Output, output)
+	}
+
+	waitForVirtHandlerWithPrHelperReadyOnNode := func(node string) {
+		ready := false
+		fieldSelector, err := fields.ParseSelector("spec.nodeName==" + string(node))
+		Expect(err).ToNot(HaveOccurred())
+		labelSelector, err := labels.Parse(fmt.Sprintf(v1.AppLabel + "=virt-handler"))
+		Expect(err).ToNot(HaveOccurred())
+		selector := metav1.ListOptions{FieldSelector: fieldSelector.String(), LabelSelector: labelSelector.String()}
+		Eventually(func() bool {
+			pods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(), selector)
+			Expect(err).ToNot(HaveOccurred())
+			if len(pods.Items) < 1 {
+				return false
+			}
+			// Virt-handler will be deployed together with the
+			// pr-helepr container
+			if len(pods.Items[0].Spec.Containers) != 2 {
+				return false
+			}
+			for _, status := range pods.Items[0].Status.ContainerStatuses {
+				if status.State.Running != nil {
+					ready = true
+				} else {
+					return false
+				}
+			}
+			return ready
+
+		}, 90*time.Second, 1*time.Second).Should(BeTrue())
+	}
+	BeforeEach(func() {
+		fgDisabled = !checks.HasFeature(virtconfig.PersistentReservation)
+		if fgDisabled {
+			tests.EnableFeatureGate(virtconfig.PersistentReservation)
```

Comment:
We  need to use "custom" `EnableFeatureGate` which will check the endpoint only after the re-deployment.